src/ros_main.cpp

```diff
@@ -134,22 +138,36 @@ int main(int argc, char **argv)
   auto rt_commander = factory.getCommander(rt_stream);
   vector<IConsumer<RTPacket> *> rt_vec{ &rt_pub };
 
-  TrajectoryFollower traj_follower(*rt_commander, local_ip, args.reverse_port, factory.isVersion3());
+  TrajectoryFollowerInterface *traj_follower(nullptr);
+
+  if (args.use_lowbandwidth_trajectory_follower && !args.use_ros_control)
+  {
+    LOG_INFO("Use low bandwidth trajectory follower");
+    traj_follower = new LowBandwidthTrajectoryFollower(*rt_commander, local_ip, args.reverse_port,factory.isVersion3());
+    // We are leaking the follower here, but it's OK as this is once-a-lifetime event
+  }
+  else
+  {
+    LOG_INFO("Use standard trajectory follower");
+    traj_follower = new TrajectoryFollower(*rt_commander, local_ip, args.reverse_port, factory.isVersion3());
+    // We are leaking the follower here, but it's OK as this is once-a-lifetime event
+  }
 
   INotifier *notifier(nullptr);
   ROSController *controller(nullptr);
   ActionServer *action_server(nullptr);
   if (args.use_ros_control)
   {
     LOG_INFO("ROS control enabled");
-    controller = new ROSController(*rt_commander, traj_follower, args.joint_names, args.max_vel_change, args.tcp_link);
+    // Note - we are sure that TrajectoryFollower is used here (see the args.use_ros_control above)
+    controller = new ROSController(*rt_commander, *((TrajectoryFollower *) traj_follower), args.joint_names, args.max_vel_change, args.tcp_link);
```

Comment:
This cast is really icky in my opinion. If `LowBandwidthTrajectoryFollower` is not possible to implement for ROS control I'd rather change `use_ros_control`, `use_lowbandwidth_trajectory_follower` and the implicit `use_action_server` option into an enum.

Reply:
It is tricky, I know. Again, the reason is that I wanted to get minimal set of changes. I actually wrote about it in my "design decisions". I think there are no fundamental reasons why ros_control would not work with LBTF. But I had no working ros_control configuration (we only used the direct setup) and implementing it also for ros control would mean creating it, making sure it works, and fixing it together with the "direct" interface. I simply had no time  for it :) and I thought that it might be a value to have it working for direct interface and once it  is proven, to ask someone who actually uses ros_control to make this (rather small) modification and test it. This would be most reasonable use of my time I think. Do you think this is the approach we might take?